src/main/kotlin/org/rust/lang/core/types/ty/TyReference.kt

```diff
@@ -7,12 +7,26 @@ package org.rust.lang.core.types.ty
 
 import org.rust.lang.core.types.infer.TypeFolder
 import org.rust.lang.core.types.infer.TypeVisitor
+import org.rust.lang.core.types.re.Re
 
-data class TyReference(val referenced: Ty, val mutability: Mutability) : Ty(referenced.flags) {
+data class TyReference(
+    val referenced: Ty,
+    val mutability: Mutability,
+    val lifetime: Re
+) : Ty(referenced.flags or lifetime.flags) {
 
-    override fun superFoldWith(folder: TypeFolder): Ty =
-        TyReference(referenced.foldWith(folder), mutability)
+    override fun superFoldWith(folder: TypeFolder): Ty = TyReference(referenced.foldWith(folder), mutability, lifetime)
 
-    override fun superVisitWith(visitor: TypeVisitor): Boolean =
-        referenced.visitWith(visitor)
+    override fun superVisitWith(visitor: TypeVisitor): Boolean = referenced.visitWith(visitor)
+
+    override fun equals(other: Any?): Boolean = when {
```

Comment:
Could you please add some comments here about why do you exclude lifetime from comparision

Reply:
I have added a comment in the code